build_runner/test/common/test_phases.dart

```diff
@@ -121,19 +117,31 @@ void checkBuild(BuildResult result,
     InMemoryAssetWriter writer,
     BuildStatus status = BuildStatus.success,
     Matcher exceptionMatcher,
-    bool writeToCache: false,
     String rootPackage}) {
   expect(result.status, status, reason: '$result');
   if (exceptionMatcher != null) {
     expect(result.exception, exceptionMatcher);
   }
 
-  var mapAssetIds = writeToCache
-      ? (AssetId id) => new AssetId(
+  final unhiddenOutputs = <String, dynamic>{};
+  final unhiddenAssets = new Set<AssetId>();
+  for (final id in outputs?.keys ?? const []) {
+    if (id.startsWith(r'$$')) {
```

Comment:
probably worth adding a comment to this function to call out this behavior

Reply:
Done